src/precice/impl/MappingContext.hpp

```diff
@@ -25,6 +25,12 @@ struct MappingContext {
 
   /// True, if data has been mapped already.
   bool hasMappedData = false;
+
+  /// Enables gradient data in the corresponding 'from' data class
+  void enableGradientData(const std::string &dataName)
```

Comment:
```suggestion
  void requireGradientData(const std::string &dataName)
```
I have the feeling "enable" is the wrong word here. Why not "require" again?